core/trino-main/src/main/java/io/trino/metadata/TypeRegistry.java

```diff
@@ -196,7 +197,12 @@ public Type getType(TypeId id)
 
     public Type fromSqlType(String sqlType)
     {
-        return getType(toTypeSignature(SQL_PARSER.createType(sqlType)));
+        try {
+            return getType(toTypeSignature(SQL_PARSER.createType(sqlType)));
+        }
+        catch (ParsingException ex) {
+            throw new TypeNotFoundException(new TypeSignature(sqlType), ex);
```

Comment:
The issue here is that `new TypeSignature(sqlType)` may throw `IllegalArgumentException`; which is not optimal. Maybe we need another exception after all.

Reply:
Or we can change `TypeNotFoundException` to allow construction with `String` and store type as `String` internally. The 
```
 public TypeSignature getType()
    {
        return type;
    }
``` 
is not used anyway so I think it can be removed